storage/innobase/xtrabackup/test/t/innodb_keyring_file.sh

```diff
@@ -2,67 +2,80 @@
 # Basic test of InnoDB encryption support
 #
 . inc/keyring_common.sh
-. inc/keyring_file.sh
-
+KEYRING_TEST_TYPES=("plugin")
+if is_server_version_higher_than 8.0.23
+then
+  KEYRING_TEST_TYPES=("plugin" "component" "both")
+fi
 function cleanup_keyring() {
 	rm -rf $keyring_file
 }
 
-test_do "ENCRYPTION='y'" "top-secret"
-test_do "ENCRYPTION='y' ROW_FORMAT=COMPRESSED" "none"
-test_do "ENCRYPTION='y' COMPRESSION='lz4'" "none"
-
-#
-# PXB-2216 - Backup changing table encryption version
-#
-MYSQLD_EXTRA_MY_CNF_OPTS="${MYSQLD_EXTRA_MY_CNF_OPTS:-""}
-innodb-buffer-pool-size=536870912
-innodb-log-file-size=536870912
-"
-
-XB_EXTRA_MY_CNF_OPTS="${XB_EXTRA_MY_CNF_OPTS:-""}
-xtrabackup-plugin-dir=${plugin_dir}
-"
-prepare_options="--xtrabackup-plugin-dir=${plugin_dir} ${keyring_args}"
-start_server
-run_cmd $MYSQL $MYSQL_ARGS test -e "SELECT @@server_uuid"
-run_cmd $MYSQL $MYSQL_ARGS test -e "CREATE TABLE tb1 (ID INT PRIMARY KEY) ENCRYPTION='Y'"
-for i in $(seq 1 10);
+for KEYRING_TYPE in "${KEYRING_TEST_TYPES[@]}";
```

Comment:
This loop will not setup for component keyring ? only plugin?

Reply:
@satya-bodapati it will set `KEYRING_TYPE` , then on keyring_common.sh -> do_test we will react accordning to the keyring type.

Reply:
sorry, I just saw the top "KEYRING_TEST_TYPES=("plugin") only :smile: 
I see multiple types for version > 8023 ? but 8023? shouldn't this be 8024?

Reply:
brain freeze moment :flushed: